### PR TITLE
Support for TChannel (except headers)

### PR DESCRIPTION
**Pull request by kriskowal**
r @yarpc/yarpc @yarpc/node

### CODE REVIEW

**Comment by breerly** [comment]
dayuuuummmm :tada:

File: example/simple/tchannel/json/server.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+var RPC = require('../../../../rpc');
```
Comment (breerly) [inline example/simple/tchannel/json/server.js]: what's with the deep upwards rel path?

Comment (kriskowal) [inline example/simple/tchannel/json/server.js]: All modules in the same package must be referenced with a relative path. Only platform modules, e.g., fs, and external packages use non-relative paths.

**Comment by abhinav** [comment]
lgtm :+1:

**Comment by breerly** [comment]
Looks like this is why the crossdock tests won't pass:

```
Attaching to yarpcnode_yarpc-go_1, yarpcnode_yarpc-java_1, yarpcnode_yarpc-node_1, yarpcnode_crossdock_1
yarpc-node_1 | module.js:327
yarpc-node_1 |     throw err;
yarpc-node_1 |     ^
yarpc-node_1 |
yarpc-node_1 | Error: Cannot find module 'tchannel'
yarpc-node_1 |     at Function.Module._resolveFilename (module.js:325:15)
yarpc-node_1 |     at Function.Module._load (module.js:276:25)
yarpc-node_1 |     at Module.require (module.js:353:17)
yarpc-node_1 |     at require (internal/module.js:12:17)
yarpc-node_1 |     at Object.<anonymous> (/src/test/subject.js:23:16)
yarpc-node_1 |     at Module._compile (module.js:409:26)
yarpc-node_1 |     at Object.Module._extensions..js (module.js:416:10)
yarpc-node_1 |     at Module.load (module.js:343:32)
yarpc-node_1 |     at Function.Module._load (module.js:300:12)
yarpc-node_1 |     at Module.require (module.js:353:17)
```

**Comment by kriskowal** [comment]
Tests now pass (ty @breerly for your investigation). This commit adds TChannel support inbound,outbound*all-encodings except for headers encoding (see #33).

Ready for an earnest review @yarpc/yarpc @yarpc/node

**Pull request merged**